lapack/gonum/dgtsv.go

```diff
@@ -0,0 +1,100 @@
+// Copyright ©2020 The Gonum Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package gonum
+
+import "math"
+
+// Dgtsv solves the equation
+//  A * X = B
+// where A is an n×n tridiagonal matrix. It uses Gaussian elimination with
+// partial pivoting. The equation Aᵀ * X = B may be solved by swapping the
+// arguments for du and dl.
+//
+// On entry, dl, d and du contain the sub-diagonal, the diagonal and the
+// super-diagonal, respectively, of A. On return, the first n-2 elements of dl
+// will be overwritten by the n-2 elements of the second super diagonal of the
```

Comment:
I'm split on documenting this. I think it's useless without the matrix L and it's practically impossible to test. Saying it's overwritten on output would be much easier without any loss.

Reply:
Besides, if nrhs is 0, correctly we should still compute the factorization which is not what the code currently does.

Reply:
> ... it's practically impossible to test

I realized that of course I could test by comparing to the result of Dgetrf but it still complicates the interface and the test for no clear benefit. If the LU factorization of A is needed, then Dgttrf can be used which I'll add soon.

> What can be done with the elements of dl with this information?

Upon a successful return, d, du and dl contain the matrix U from the LU factorization of A. I don't see any use for U without L, I think this is more of a side-effect of using the slices for storing the factorization somewhere. And if the factorization is needed, then Dgttrf will be available. For me Dgtsv is just a convenience function for one-time solution of a linear system without getting the factorization explicitly. I plan to use Dgtsv in mat.TridiagDense.SolveTo and Dgttrf in mat.TridiagLU.Factorize.

In total, I've change the documentation to say that dl, d and du will be overwritten without specifying how.